### PR TITLE
Filter all `DeprecationWarning`'s by `ArrowTable.to_pandas()`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR filters all `DeprecationWarning`'s that are being originated by `ArrowTable.to_pandas`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py [hunk 1]
```diff
@@ -22,9 +22,17 @@
...
 import cudf
+from cudf.core._compat import PANDAS_GE_220
 from cudf.testing._utils import assert_eq
 from cudf.testing.dataset_generator import rand_dataframe
 
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
```
Comment (mroeschke) [inline python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py:33]: cudf never deals with the pandas block manager right? If not, I would just globally ignore this in `python/cudf/cudf/tests/pytest.ini` with `ignore:Passing a BlockManager to DataFrame is deprecated:DeprecationWarning`

Comment (galipremsagar) [reply python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py:33]: We don't deal with it, it shows up every time when we do:
```
arrow_table = pa.table(...)
arrow_table.to_pandas()
```

File: python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py [hunk 2]
```diff
@@ -22,9 +22,17 @@
 import pytest
 
 import cudf
+from cudf.core._compat import PANDAS_GE_220
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py:25]:
```suggestion
```

File: python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py [hunk 3]
```diff
@@ -22,9 +22,17 @@
...
 from cudf.testing._utils import assert_eq
 from cudf.testing.dataset_generator import rand_dataframe
 
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
+    )
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py:35]:
```suggestion
```

File: python/cudf/cudf/tests/test_dataframe.py [hunk 4]
```diff
@@ -25,7 +25,12 @@
 
 import cudf
 from cudf.api.extensions import no_default
-from cudf.core._compat import PANDAS_GE_200, PANDAS_GE_210, PANDAS_LT_203
+from cudf.core._compat import (
+    PANDAS_GE_200,
+    PANDAS_GE_210,
+    PANDAS_GE_220,
+    PANDAS_LT_203,
+)
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_dataframe.py:33]:
```suggestion
from cudf.core._compat import PANDAS_GE_200, PANDAS_GE_210, PANDAS_LT_203
```

File: python/cudf/cudf/tests/test_dataframe.py [hunk 5]
```diff
@@ -43,7 +48,17 @@
...
+
+if PANDAS_GE_220:
+    pytestmark.append(
+        pytest.mark.filterwarnings(
+            "ignore",
+            category=DeprecationWarning,
+            message="Passing a BlockManager to DataFrame is deprecated",
+        )
+    )
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_dataframe.py:61]:
```suggestion
pytestmark = pytest.mark.spilling
```

File: python/cudf/cudf/tests/test_datetime.py [hunk 6]
```diff
@@ -37,6 +37,13 @@
     operator.ne,
 ]
 
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
+    )
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_datetime.py:46]:
```suggestion
```

File: python/cudf/cudf/tests/test_feather.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2018-2023, NVIDIA CORPORATION.
+# Copyright (c) 2018-2024, NVIDIA CORPORATION.
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_feather.py:1]:
```suggestion
# Copyright (c) 2018-2023, NVIDIA CORPORATION.
```

File: python/cudf/cudf/tests/test_feather.py [hunk 8]
```diff
@@ -9,8 +9,16 @@
 import pytest
 
 import cudf
+from cudf.core._compat import PANDAS_GE_220
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_feather.py:12]:
```suggestion
```

File: python/cudf/cudf/tests/test_feather.py [hunk 9]
```diff
@@ -9,8 +9,16 @@
...
 from cudf.testing._utils import NUMERIC_TYPES, assert_eq
 
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
+    )
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_feather.py:21]:
```suggestion
```

File: python/cudf/cudf/tests/test_groupby.py [hunk 10]
```diff
@@ -20,7 +20,7 @@
 import cudf
 from cudf import DataFrame, Series
 from cudf.api.extensions import no_default
-from cudf.core._compat import PANDAS_GE_200, PANDAS_GE_210
+from cudf.core._compat import PANDAS_GE_200, PANDAS_GE_210, PANDAS_GE_220
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_groupby.py:23]:
```suggestion
from cudf.core._compat import PANDAS_GE_200, PANDAS_GE_210
```

File: python/cudf/cudf/tests/test_groupby.py [hunk 11]
```diff
@@ -40,6 +40,13 @@
 _tomorrow = np.int64(_tomorrow.astype("datetime64[ns]"))
 _index_type_aggs = {"count", "idxmin", "idxmax", "cumcount"}
 
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
+    )
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_groupby.py:49]:
```suggestion
```

File: python/cudf/cudf/tests/test_index.py [hunk 12]
```diff
@@ -15,7 +15,7 @@
 import cudf
 from cudf.api.extensions import no_default
 from cudf.api.types import is_bool_dtype
-from cudf.core._compat import PANDAS_GE_200
+from cudf.core._compat import PANDAS_GE_200, PANDAS_GE_220
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_index.py:18]:
```suggestion
from cudf.core._compat import PANDAS_GE_200
```

File: python/cudf/cudf/tests/test_index.py [hunk 13]
```diff
@@ -39,6 +39,13 @@
 )
 from cudf.utils.utils import search_range
 
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
+    )
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_index.py:48]:
```suggestion
```

File: python/cudf/cudf/tests/test_orc.py [hunk 14]
```diff
@@ -13,6 +13,7 @@
 import pytest
 
 import cudf
+from cudf.core._compat import PANDAS_GE_220
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_orc.py:16]:
```suggestion
```

File: python/cudf/cudf/tests/test_orc.py [hunk 15]
```diff
@@ -22,13 +23,24 @@
...
+]
+
+if PANDAS_GE_220:
+    pytestmark.append(
+        pytest.mark.filterwarnings(
+            "ignore",
+            category=DeprecationWarning,
+            message="Passing a BlockManager to DataFrame is deprecated",
+        )
+    )
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_orc.py:43]:
```suggestion
# Removal of these deprecated features is no longer imminent. They will not be
# removed until a suitable alternative has been implemented. As a result, we
# also do not want to stop testing them yet.
# https://github.com/rapidsai/cudf/issues/11519
pytestmark = pytest.mark.filterwarnings(
        "ignore:(num_rows|skiprows) is deprecated and will be removed."
)
```

File: python/cudf/cudf/tests/test_parquet.py [hunk 16]
```diff
@@ -21,7 +21,7 @@
 from pyarrow import fs as pa_fs, parquet as pq
 
 import cudf
-from cudf.core._compat import PANDAS_GE_200
+from cudf.core._compat import PANDAS_GE_200, PANDAS_GE_220
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_parquet.py:24]:
```suggestion
from cudf.core._compat import PANDAS_GE_200
```

File: python/cudf/cudf/tests/test_parquet.py [hunk 17]
```diff
@@ -34,6 +34,13 @@
     set_random_null_mask_inplace,
 )
 
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
+    )
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_parquet.py:43]:
```suggestion
```

File: python/cudf/cudf/tests/test_rolling.py [hunk 18]
```diff
@@ -8,10 +8,17 @@
 import pytest
 
 import cudf
-from cudf.core._compat import PANDAS_GE_200
+from cudf.core._compat import PANDAS_GE_200, PANDAS_GE_220
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_rolling.py:11]:
```suggestion
from cudf.core._compat import PANDAS_GE_200
```

File: python/cudf/cudf/tests/test_rolling.py [hunk 19]
```diff
@@ -8,10 +8,17 @@
...
 from cudf.testing._utils import assert_eq
 from cudf.testing.dataset_generator import rand_dataframe
 
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
+    )
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_rolling.py:21]:
```suggestion
```

File: python/cudf/cudf/tests/test_s3.py [hunk 20]
```diff
@@ -12,13 +12,20 @@
 from fsspec.core import get_fs_token_paths
 
 import cudf
+from cudf.core._compat import PANDAS_GE_220
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_s3.py:15]:
```suggestion
```

File: python/cudf/cudf/tests/test_s3.py [hunk 21]
```diff
@@ -12,13 +12,20 @@
...
 boto3 = pytest.importorskip("boto3")
 s3fs = pytest.importorskip("s3fs")
 
 ThreadedMotoServer = pytest.importorskip("moto.server").ThreadedMotoServer
+if PANDAS_GE_220:
+    pytestmark = pytest.mark.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="Passing a BlockManager to DataFrame is deprecated",
+    )
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_s3.py:28]:
```suggestion
```

File: python/cudf/cudf/tests/test_orc.py [hunk 22]
```diff
@@ -27,7 +27,7 @@
 # also do not want to stop testing them yet.
 # https://github.com/rapidsai/cudf/issues/11519
 pytestmark = pytest.mark.filterwarnings(
-    "ignore:(num_rows|skiprows) is deprecated and will be removed."
+        "ignore:(num_rows|skiprows) is deprecated and will be removed."
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_orc.py:30]:
```suggestion
      "ignore:(num_rows|skiprows) is deprecated and will be removed."
```

File: python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py]: Done, moved the filter to `pytest.ini`

**Review by mroeschke — APPROVED** [review]

File: python/cudf/cudf/tests/pytest.ini [hunk 23]
```diff
@@ -8,3 +8,4 @@ filterwarnings =
     error
     ignore:::.*xdist.*
     ignore:::.*pytest.*
+    ignore:Passing a BlockManager to DataFrame is deprecated:DeprecationWarning
```
Comment (mroeschke) [inline python/cudf/cudf/tests/pytest.ini:11]: In a follow up, could you add a comment here that this comes from pyarrow starting with pandas 2.2?

Comment (galipremsagar) [reply python/cudf/cudf/tests/pytest.ini:11]: Done, in this pr itself 👍

**Pull request merged**